src/Core/Ast/Parser/TypeResolver.php

```diff
@@ -61,127 +61,167 @@ public function resolvePHPParserTypes(TypeScope $typeScope, NodeAbstract ...$nod
-        }
-        if ($type instanceof UnionTypeNode || $type instanceof IntersectionTypeNode) {
-            return array_merge([], ...array_map(fn (TypeNode $typeNode) => $this->resolvePHPStanDocParserType($typeNode, $typeScope, $templateTypes), $type->types));
-        }
-        if ($type instanceof GenericTypeNode) {
-            $preType = 'list' === $type->type->name ? [] : $this->resolvePHPStanDocParserType($type->type, $typeScope, $templateTypes);
-
-            return array_merge($preType, ...array_map(fn (TypeNode $typeNode) => $this->resolvePHPStanDocParserType($typeNode, $typeScope, $templateTypes), $type->genericTypes));
-        }
-        if ($type instanceof ArrayShapeNode) {
-            return array_merge([], ...array_map(fn (ArrayShapeItemNode $itemNode) => $this->resolvePHPStanDocParserType($itemNode->valueType, $typeScope, $templateTypes), $type->items)
-            );
-        }
-        if ($type instanceof CallableTypeNode) {
-            return array_merge(
-                $this->resolvePHPStanDocParserType($type->returnType, $typeScope, $templateTypes),
-                ...array_map(fn (CallableTypeParameterNode $parameterNode) => $this->resolvePHPStanDocParserType($parameterNode->type, $typeScope, $templateTypes), $type->parameters)
-            );
-        }
-
-        return $this->resolveString((string) $type, $typeScope);
+        return match (true) {
+            $type instanceof IdentifierTypeNode => in_array($type->name, $templateTypes, true) ? [] : $this->resolveString($type->name, $typeScope),
+            $type instanceof ConstTypeNode && $type->constExpr instanceof ConstFetchNode => $this->resolveString($type->constExpr->className, $typeScope),
+            $type instanceof NullableTypeNode => $this->resolvePHPStanDocParserType($type->type, $typeScope, $templateTypes),
+            $type instanceof ArrayTypeNode => $this->resolvePHPStanDocParserType($type->type, $typeScope, $templateTypes),
+            $type instanceof UnionTypeNode || $type instanceof IntersectionTypeNode => $this->resolveVariableType($type, $typeScope, $templateTypes),
+            $type instanceof GenericTypeNode => $this->resolveGeneric($type, $typeScope, $templateTypes),
+            $type instanceof ArrayShapeNode => $this->resolveArray($type, $typeScope, $templateTypes),
+            $type instanceof CallableTypeNode => $this->resolveCallable($type, $typeScope, $templateTypes),
+            default => $this->resolveString((string) $type, $typeScope)
+        };
     }
 
     /**
      * @return string[]
      */
-    protected function resolveString(string $type, TypeScope $nameScope): array
+    private function resolveString(string $type, TypeScope $nameScope): array
     {
-        $context = new Context($nameScope->namespace, $nameScope->getUses());
         try {
+            $context = new Context($nameScope->namespace, $nameScope->getUses());
```

Comment:
Everything inside the try-catch could hide possible bugs, at this point I don't know if it is critical or not.

Reply:
Nothing else that was added to the `try` block throws exceptions.